feature/settings/src/main/java/com/google/jetpackcamera/settings/ui/SettingsComponents.kt

```diff
@@ -634,6 +752,34 @@ fun SingleChoiceSelector(
     }
 }
 
+@Composable
+@ReadOnlyComposable
+fun disabledRationaleString(disabledRationale: DisabledRationale): String {
+    val context = LocalContext.current
+    return when (disabledRationale) {
+        is DisabledRationale.DeviceUnsupportedRationale -> context.getString(
+            disabledRationale.reasonTextResId,
+            disabledRationale.affectedSettingName
+        )
+
+        is DisabledRationale.FpsUnsupportedRationale -> context.getString(
+            disabledRationale.reasonTextResId,
+            disabledRationale.affectedSettingName,
+            disabledRationale.currentFps
+        )
+
+        is DisabledRationale.LensUnsupportedRationale -> context.getString(
+            disabledRationale.reasonTextResId,
+            disabledRationale.affectedSettingName
+        )
+
+        is DisabledRationale.StabilizationUnsupportedRationale -> context.getString(
+            disabledRationale.reasonTextResId,
+            disabledRationale.affectedSettingName
+        )
+    }
+}
+
```

Comment:
nit: since this is a `@Composable`, you don't need to use `LocalContext.current`. You can just use `stringResource`:
```
@Composable
@ReadOnlyComposable
fun disabledRationaleString(disabledRationale: DisabledRationale): String =
    when (disabledRationale) {
        is DisabledRationale.DeviceUnsupportedRationale -> stringResource(
            disabledRationale.reasonTextResId,
            disabledRationale.affectedSettingName
        )

        is DisabledRationale.FpsUnsupportedRationale -> stringResource(
                disabledRationale.reasonTextResId,
                disabledRationale.affectedSettingName,
                disabledRationale.currentFps
        )

        is DisabledRationale.LensUnsupportedRationale -> stringResource(
            disabledRationale.reasonTextResId,
            disabledRationale.affectedSettingName
        )

        is DisabledRationale.StabilizationUnsupportedRationale -> stringResource(
            disabledRationale.reasonTextResId,
            disabledRationale.affectedSettingName
        )
    }
```